python/cog/server/worker.py

```diff
@@ -141,6 +141,19 @@ def _wait(
             )
 
 
+class LockedConn:
+    def __init__(self, conn: Connection) -> None:
+        self.conn = conn
+        self._lock = _spawn.Lock()
+
+    def send(self, obj: Any) -> None:
+        with self._lock:
+            self.conn.send(obj)
+
+    def recv(self) -> Any:
+        return self.conn.recv()
```

Comment:
Just confirming, `recv` doesn't require a lock, right?

Reply:
it's only used from one place at a time and probably doesn't require a lock anyawy